scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/Op.scala

```diff
@@ -155,7 +156,16 @@ object Op extends Serializable {
         for {
           l <- lrdd
           r <- rrdd
-        } yield widen[A](l) ++ widen[A](r)
+        } yield {
+          val merged = widen[A](l) ++ widen[A](r)
+          val partitions = merged.getNumPartitions
+          val newPartitions = pc(partitions)
+          if (partitions != newPartitions) {
```

Comment:
`partitions > newPartitions`
is likely the intent here, coalesce up doesn't make too much sense